app/models/order_detail.rb

```diff
@@ -942,7 +942,7 @@ def clear_statement
 
   def update_fulfilled_at_on_resolve
     if problem_changed? && !problem_order?
-      self.fulfilled_at = time_data.actual_end_at
+      self.fulfilled_at = time_data.actual_end_at if time_data.actual_end_at
```

Comment:
There's probably a weird edge case where `actual_end_at` might change, so maybe `||=` doesn't make sense. I think what you have sounds reasonable. Hopefully when you're writing specs you don't run into anything funky.